packages/plugin-js-packages/src/lib/runner/audit/transform.ts

```diff
@@ -0,0 +1,81 @@
+import type { AuditOutput, Issue, IssueSeverity } from '@code-pushup/models';
+import { objectToEntries } from '@code-pushup/utils';
+import {
+  PackageAuditLevel,
+  PackageDependency,
+  packageAuditLevels,
+} from '../../config';
+import { NpmAuditResultJson, Vulnerabilities } from './types';
+
+export function auditResultToAuditOutput(
+  result: NpmAuditResultJson,
+  dependenciesType: PackageDependency,
+  auditLevelMapping: Record<PackageAuditLevel, IssueSeverity>,
+): AuditOutput {
+  const issues = vulnerabilitiesToIssues(
+    result.vulnerabilities,
+    auditLevelMapping,
+  );
+  return {
+    slug: `npm-audit-${dependenciesType}`,
+    score: result.metadata.vulnerabilities.total === 0 ? 1 : 0,
+    value: result.metadata.vulnerabilities.total,
+    displayValue: vulnerabilitiesToDisplayValue(
+      result.metadata.vulnerabilities,
+    ),
+    ...(issues.length > 0 && { details: { issues } }),
+  };
+}
+
+export function vulnerabilitiesToDisplayValue(
+  vulnerabilities: Record<PackageAuditLevel | 'total', number>,
+): string {
+  if (vulnerabilities.total === 0) {
+    return 'passed';
+  }
+
+  const displayValue = packageAuditLevels
+    .map(level =>
+      vulnerabilities[level] > 0 ? `${vulnerabilities[level]} ${level}` : '',
+    )
+    .filter(text => text !== '')
+    .join(', ');
+  return `${displayValue} ${
+    vulnerabilities.total === 1 ? 'vulnerability' : 'vulnerabilities'
+  }`;
+}
+
+export function vulnerabilitiesToIssues(
+  vulnerabilities: Vulnerabilities,
+  auditLevelMapping: Record<PackageAuditLevel, IssueSeverity>,
+): Issue[] {
+  if (Object.keys(vulnerabilities).length === 0) {
+    return [];
+  }
+
+  return objectToEntries(vulnerabilities).map<Issue>(([, detail]) => {
+    // Advisory details via can refer to another vulnerability
+    // For now, only direct context is supported
+    if (
+      Array.isArray(detail.via) &&
+      detail.via.length > 0 &&
+      typeof detail.via[0] === 'object'
+    ) {
+      return {
+        message: `${detail.name} dependency has a vulnerability "${
+          detail.via[0].title
+        }" for versions ${detail.range}. Fix is ${
+          detail.fixAvailable ? '' : 'not '
+        }available. More information [here](${detail.via[0].url})`,
+        severity: auditLevelMapping[detail.severity],
+      };
+    }
+
+    return {
+      message: `${detail.name} dependency has a vulnerability for versions ${
+        detail.range
+      }. Fix is ${detail.fixAvailable ? '' : 'not '}available.`,
+      severity: auditLevelMapping[detail.severity],
+    };
+  });
+}
```

Comment:
Suggestions:
- use markdown inline typograhy to make key info pop out
- include audit level
- move advisory title to end of message and merge with URL
  - inserting mid-sentence makes the message hard to read
  - this way we can include everything in `via`, as long as we filter out duplicates
- include `effects` (so I know which direct depedency pulls it in)
  - ideally the range as well (that's what I need) 
  - `npm audit`'s text output actually merges vulnerabilities linked through `effects` and `via` - perhaps we could do the same?
     ![image](https://github.com/code-pushup/cli/assets/34691111/4e4f464e-832b-483d-8acb-2cec8fef135b)


Before:
> @babel/traverse dependency has a vulnerability "Babel vulnerable to arbitrary code execution when compiling specifically crafted malicious code" for versions <7.23.2. Fix is available. More information [here](https://github.com/advisories/GHSA-67hx-6x53-jw92)

After:
> `@babel/traverse` package has a **critical**-severity vulnerability in versions **<7.23.2**. Fix is available. Affects direct dependency `@trivago/prettier-plugin-sort-imports` in versions **2.0.0-0 - 4.2.0**. More info: [Babel vulnerable to arbitrary code execution when compiling specifically crafted malicious code](https://github.com/advisories/GHSA-67hx-6x53-jw92)

Reply:
Ad cross-reference context: That is OOS for this PR, I am aware of it. Would like to include the `effects` as OOS as well here tbh.
![image](https://github.com/code-pushup/cli/assets/6306671/85e0e1d2-cc61-4e38-a149-6997bd038432)

Reply:
Fair enough, it's already much improved now :+1: